### PR TITLE
Ensure the software context is ready

### PR DESCRIPTION
**Pull request by imobachgs**
## Problem

For a short time (a fraction of a second), the overview page is shown even if you have not selected a product. It's not a problem at all for a human, but something nasty for automated testing.

## Solution

Do not load the application until the software context is ready.

## Testing

- Adapted a new unit test
- Tested manually

### CODE REVIEW

File: web/src/App.test.jsx [hunk 1]
```diff
@@ -63,13 +75,29 @@ describe("App", () => {
...
     // setting a cookie with already expired date removes it
     document.cookie = "CockpitLang=; path=/; expires=" + new Date(0).toUTCString();
   });
 
+  describe("when the software context is not initialized", () => {
+    beforeEach(() => {
+      mockProducts = undefined;
+    });
+
+    it("renders the LoadingEnvironment theme", async () => {
```
Comment (dgdavid) [inline web/src/App.test.jsx:95]: NP

```suggestion
    it("renders the LoadingEnvironment screen", async () => {
```

**Review by dgdavid — APPROVED** [review]
LGTM, but please put something meaningful in the PR description

**Comment by imobachgs** [comment]
> LGTM, but please put something meaningful in the PR description

Sure! I am just trying to figure out why Playwright tests are failing (they work on my machine:tm:).

**Review by joseivanlopez — APPROVED** [review]
LGTM

**Pull request merged**